Modules/Devices/src/Devices.Application/PushNotifications/Commands/UpdateDeviceRegistration/Handler.cs

```diff
@@ -16,16 +18,22 @@ public class Handler : IRequestHandler<UpdateDeviceRegistrationCommand, UpdateDe
     private readonly DeviceId _activeDevice;
     private readonly IdentityAddress _activeIdentity;
     private readonly IPushNotificationRegistrationService _pushRegistrationService;
+    private readonly IIdentitiesRepository _identitiesRepository;
 
-    public Handler(IPushNotificationRegistrationService pushRegistrationService, IUserContext userContext)
+    public Handler(IPushNotificationRegistrationService pushRegistrationService, IUserContext userContext, IIdentitiesRepository identitiesRepository)
     {
         _pushRegistrationService = pushRegistrationService;
         _activeIdentity = userContext.GetAddress();
         _activeDevice = userContext.GetDeviceId();
+        _identitiesRepository = identitiesRepository;
     }
 
     public async Task<UpdateDeviceRegistrationResponse> Handle(UpdateDeviceRegistrationCommand request, CancellationToken cancellationToken)
     {
+        var identity = await _identitiesRepository.FindByAddress(_activeIdentity, CancellationToken.None) ?? throw new NotFoundException(nameof(Identity));
+
+        identity.EnsureIdentityIsToBeDeleted();
```

Comment:
Why do you forbid the update of the PNS handle in case the identity is in status ToBeDeleted?